Util/Tokenization/src/main/java/org/tribuo/util/tokens/impl/SplitCharactersTokenizer.java

```diff
@@ -36,34 +36,64 @@
  * that is considered whitespace by {@link Character#isWhitespace(char)}.
  * @author Philip Ogren
  */
-public class SplitCharactersTokenizer implements Tokenizer {
+public class SplitCharactersTokenizer extends SplitFunctionTokenizer {
 
     public static final char[] DEFAULT_SPLIT_CHARACTERS = new char[]{'*', '(', ')', '&', '[', ']', '{', '}', '`',
             '\'', '|', ';', ':', '\\', '!', '-', '?'};
     public static final char[] DEFAULT_SPLIT_EXCEPTING_IN_DIGITS_CHARACTERS = new char[]{'.', ',', '/',};
 
+	public static class SplitCharactersSplitterFunction implements SplitFunction {
+
+	    private char[] splitCharacters = DEFAULT_SPLIT_CHARACTERS;
+
+	    private char[] splitXDigitsCharacters = DEFAULT_SPLIT_EXCEPTING_IN_DIGITS_CHARACTERS;
+
+		public SplitCharactersSplitterFunction(char[] splitCharacters, char[] splitXDigitsCharacters) {
+			this.splitCharacters = splitCharacters;
+			this.splitXDigitsCharacters = splitXDigitsCharacters;
+		}
+		
+		public SplitResult apply(int codepoint, int index, CharSequence cs) {
+            if(isSplitCharacter((char)codepoint)) {
```

Comment:
The apply method should have an `@Override` annotation.  It is an `int codepoint` because that's how I defined the interface to work with a tokenizer that operates on codepoints. This tokenizer was retro-fitted to extend SplitFunctionTokenizer and was previously a character-by-character tokenizer.  I suppose I should revert this class back to its original definition since this is not likely a backwards compatible change.

Reply:
This PR introduces the `SplitFunctionTokenizer` and it was useful to update `SplitCharactersTokenizer` for testing purposes.

Reply:
If it is interface compatible then I'm fine with it, I was just wondering why it wasn't restricted to `char`. I guess `int` covers more of the UTF-16 space.

Reply:
that's the idea.  There's always 4-byte emojis lurking around and we definitely don't want to be splitting those down the middle.